softvis3d-frontend/src/components/ui/selectbox/SelectBox.tsx

```diff
@@ -67,19 +77,39 @@ export default class SelectBox extends React.Component<SelectBoxProps, any> {
         );
     }
 
+    private addValue(v: SelectOptionValue) {
+        this.values[v.getValue()] = v;
+    }
+
+    private clearValues() {
+        this.values = {};
+    }
+
     private renderChildren(): Array<React.Component<any, any>> {
+        this.clearValues();
+        const ref = (o: SelectOption | null) => {
+            if (!o) {
+                this.clearValues();
+            } else {
+                const v: SelectOptionValue = o.props.value;
+                this.addValue(v);
+            }
+        };
+
         return React.Children.map<any>(
             (this.props.children as Array<SelectOption|SelectGroup>),
             (child: React.ReactElement<any>) => {
                 if (child.type === SelectOption) {
                     return React.cloneElement(child, {
                         checked: child.props.value === this.props.value,
-                        disabled: this.props.disabled || child.props.disabled
+                        disabled: this.props.disabled || child.props.disabled,
+                        ref
                     });
                 } else if (child.type === SelectGroup) {
                     return React.cloneElement(child, {
                         selectedValue: this.props.value,
-                        disabled: this.props.disabled || child.props.disabled
+                        disabled: this.props.disabled || child.props.disabled,
+                        optionRef: ref
                     });
                 } else {
                     return child;
```

Comment:
Luckily for me it's not as bad for me in this situation: I will just render the given child without manipulating it. I'll a have a look at the StatusActionQueue separately though (with #60).